datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mappers/DatasetUpdateInputMapper.java

```diff
@@ -35,62 +21,23 @@ public Dataset apply(@Nonnull final DatasetUpdateInput datasetUpdateInput) {
         final Dataset result = new Dataset();
 
         if (datasetUpdateInput.getOwnership() != null) {
-            final Ownership ownership = new Ownership();
-            ownership.setOwners(new OwnerArray(datasetUpdateInput.getOwnership().getOwners()
-                    .stream()
-                    .map(this::mapOwner)
-                    .collect(Collectors.toList())));
-            result.setOwnership(ownership);
+            result.setOwnership(OwnershipUpdateMapper.map(datasetUpdateInput.getOwnership()));
         }
 
         if (datasetUpdateInput.getDeprecation() != null) {
             final DatasetDeprecation deprecation = new DatasetDeprecation();
             deprecation.setDeprecated(datasetUpdateInput.getDeprecation().getDeprecated());
-            deprecation.setDecommissionTime(datasetUpdateInput.getDeprecation().getDecommissionTime());
+            if (datasetUpdateInput.getDeprecation().getDecommissionTime() != null) {
```

Comment:
Is this null check required? If decommission time = null, then we will just set the value of DatasetDeprecation.decomissionTime to null also

Reply:
In Case of Mutations if the user does not provide `decommisionTime` in the input then trying to read it would throw an NPE.
It would work in case of queries but for mutations null checks needs to be made for Optional fields.